### DISCUSSION TITLE
Dual screen

### DISCUSSION THREAD

**Discussion by erikdubois** [elkowar/eww#592] [Q&A]
Is there an example of code where we can have the same or similar bar on the second monitor?
Did not find it yet?

**Comment by oldwomanjosiah** [comment]
Not official or anything, but my in my personal configs I did it [like this](https://github.com/oldwomanjosiah/dots/blob/47186bae386ae0ba6ab7e46799d1c473f1335870/.config/eww/eww.yuck#L17).

Basically, I'd just abstract out the parts of the config that apply to both into separate widgets, which you then use in both.

**Reply by erikdubois** [reply]
thanks for your answer  - until now I was unsuccessfull in making it work

**Reply by oldwomanjosiah** [reply]
No problem! Were you able to get it to work like you were expecting?

**Reply by erikdubois** [reply]
partially working
but the workspaces are not ok now
It makes no sense now when clicking on the workspace numbers

**Reply by oldwomanjosiah** [reply]
What's happening, and can you provide your current configuration?

**Reply by erikdubois** [reply]
This is where I will post my experimental Fancy theme.

https://github.com/erikdubois/fancy

We start the eww with the up file from leftwm

`# EWW
eww daemon

#single screen
#for i in {0..0}

#dual screen
for i in {0..1}

do
   eww open bar$i -c $SCRIPTPATH/eww
done
eww update user=$(whoami) -c $SCRIPTPATH/eww
`
The bar is bar0 and bar1. I can tell them to go monitor 0 and 1. But when I start clicking on the numbers then it does not make sense.
We need to make the workspaces a separate thing on each monitor.

Do we need this update user line for eww?
`eww update user=$(whoami) -c $SCRIPTPATH/eww`

For some reason my bar is now fully transparent when I use two monitors. I can change what I want in the scss it will not change. I looked into picom as well. That is for another day. This does not happen with one monitor and this code - https://github.com/arcolinux/arcolinux-leftwm-theme-fancy

Transparent look (dual monitor code - experimental code)
![transparen](https://user-images.githubusercontent.com/10594806/194123068-5eb19130-028e-4699-b9bb-7275ef1c6049.png)

Single monitor (arcolinux code)
![oldcode](https://user-images.githubusercontent.com/10594806/194123455-d3c5b991-6a1e-4e9b-a493-50f9457bbed8.png)

**Reply by oldwomanjosiah** [reply]
What behavior is happening when you click on one of the workspace buttons (and what are you expecting)?

**Comment by arcolinuxz** [comment]
Tried something new - unsuccesfull
this time with wmcrtl
could you have a look
it is in the folder eww-wmctrl
https://github.com/erikdubois/fancy/tree/main/eww-wmctrl
I am trying to tell him that one bar and workspaces is on one monitor and another bar is on another monitor

**Reply by arcolinuxz** [reply]
when I click on workspace 1 on monitor 0 the workspace1 of monitor 1 activates for example
workspaces are not per monitor
they are used over the two monitors
On leftwm that is seperate with the use of two polybars on each monitor.

**Reply by erikdubois** [reply]
made a video of a dual screen to try and explain and show it

https://youtu.be/_DH_G-qj848

**Reply by oldwomanjosiah** [reply]
Oh I see, that is kinda strange! On i3, I get a similar (ish) behavior in that workspaces are "assigned" to an output. When I click on one of my workspace indicators, it switches _that workspace's assigned output_ to that workspace.

That seems possibly consistent with what you are seeing, but if I am understanding what you want it to do instead, you could probably have `wmctrl` do:
- Move desktop `n` to the output for the bar  
  (I don't know `wmctrl` enough to help with this, unfortunately)
- Switch to desktop `n`

**Comment by WilfSilver** [comment]
Hey just found this thought I'd mention for anyone else who finds this in the future, https://github.com/elkowar/eww/pull/431 seems to do what you are looking for by adding in the ability to give a bar an `id` when creating a window so you can have the same window open with different ids (and then have parameters which alter the config of the window slightly).

You can also take this further with [my other branch](https://github.com/WilfSilver/eww/tree/per_window_variables) which adds the ability to set variables to be created per window, however this is not an official implementation as we are still waiting on https://github.com/elkowar/eww/issues/324 for a better + more robust solution.

I currently use this for my setup to have the bars autogenerated on multiple screens with a simple script and it has been working quite well so far.
